tao_compiler/mlir/disc/transforms/shape_utils.cc

```diff
@@ -501,6 +501,22 @@ LogicalResult ShapeAnalysis::buildBlockDimValueMap(Block* block) {
       }
       mayCreateOrMapConstInt(operand);
       mayCreateOrMapConstInt(result);
+    } else if (isa<mhlo::DynamicReshapeOp, mhlo::DynamicBroadcastInDimOp>(op)) {
```

Comment:
is this related with the optimization of Gemm -> BatchGemm?

Reply:
It is not the logic about batching. The lack of the functions here makes the batching process fail in some cases due to failed dim-equal analysis. I will split it into a separate PR.

Reply:
It has been moved to a separate PR: https://github.com/alibaba/BladeDISC/pull/179.